src/public/components/pretty-stringify.js

```diff
@@ -0,0 +1,7 @@
+import K, * as U from "karet.util"
+import React     from "karet"
+
+export const PrettyStringify = ( { space, obj } ) =>
+  <pre className="pretty-stringify">
+    { U.lift( x => JSON.stringify( x, null, Number( space ) ) )( obj ) }
+  </pre>
```

Comment:
How about this:
```js
export default ( { value, replacer = null, space } ) =>
  <pre className="pretty-stringify">
    { U.lift( JSON.stringify )( value, replacer, U.ifElse( isNaN, _ => 2, Number )( space ) ) }
  </pre>
```
This works whether I pass in `space="2"`, `space={ 2 }`, `space={ U.atom( 2 ) }`, nothing at all or a non numeric value.